### PR TITLE
Add label and separators for Menu

### PR DESCRIPTION
**Pull request by ravicious**
This PR adds a label and a separator component. Those two can be used to create sections within the menu. I will need them to [show ports for multi-port TCP apps in Connect](https://github.com/gravitational/teleport/blob/master/rfd/0182-multi-port-tcp-app-access.md#ui). FWIW, these were not consulted with the design team.

I also refactored the story (to make it easier to add more stuff in an upcoming PR) and `MenuList` (because I was getting some weird TS errors when the component was written in JS). Also, `Flex` now accepts `inline` as a prop to use `display: inline-flex`.

https://localhost:9002/?path=/story/design-menu--menu-items

label-and-separator

ports in Connect

### CODE REVIEW

File: web/packages/design/src/Menu/Menu.story.tsx [hunk 1]
```diff
@@ -100,41 +137,51 @@ export const IconExample = () => (
...
 
-class SimpleMenu extends React.Component {
-  state = {
-    anchorEl: null,
-  };
+const SimpleMenu = (
+  props: PropsWithChildren<{
+    text: string;
+    anchorOrigin?: any;
+    transformOrigin?: any;
```
Comment (ryanclark) [inline web/packages/design/src/Menu/Menu.story.tsx:144]: Can we avoid these `any`s?

**Comment by ravicious** [comment]
@rudream @ryanclark Ping.

**Review by ryanclark — APPROVED** [review]
Sorry was OOO yesterday

**Comment by ravicious** [comment]
I screwed up and I merged #49626 into this one… FWIW it was reviewed by two people.

**Review by gzdunek — APPROVED** [review]
Looks good. 
Thanks for rewriting the component to TS 👍

**Pull request merged**